packages/qwik-city/src/middleware/request-handler/resolve-request-handlers.ts

```diff
@@ -434,7 +434,13 @@ function csrfCheckMiddleware(requestEv: RequestEvent) {
   if (isForm) {
     const inputOrigin = requestEv.request.headers.get('origin');
     const origin = requestEv.url.origin;
-    const forbidden = inputOrigin !== origin;
+    let forbidden = inputOrigin !== origin;
+
+    // fix https://github.com/QwikDev/qwik/issues/7688
+    if (origin.startsWith('https://') && inputOrigin?.replace('http://', 'https://') === origin) {
```

Comment:
```ts
    if (forbidden && origin.startsWith('https://') && inputOrigin?.slice(4) === origin.slice(5)) {
```

Reply:
it's a little bit less readable but ok, I'll change it